### ISSUE TITLE
Where is comparing raw noise and golden noise

### ISSUE THREAD

**Issue by sherlockma11** [xie-lab-ml/Golden-Noise-for-Diffusion-Models#18]
Your paper mentions that the data collection process involves comparing raw noise and golden noise, with only the latter being retained if it scores higher. However, your code does not seem to include this part. 
Furthermore, the HPSV2 scores in your code are normalized to 1, so the scores saved in the data collection code are all 1, which is meaningless!

**Comment by Klayand** [comment]
Thanks for your questions!

1. "data collection process involves comparing raw noise and golden noise"

We perform this operation in :

<img width="1271" height="818" alt="Image" src="https://github.com/user-attachments/assets/7a9bb45f-9819-4e9c-99f0-75be8d1bc033" />

2. "the HPSV2 scores in your code are normalized to 1"

We do not normalize the score, we just x100 for the consistency of performance.

**Comment by Klayand** [comment]
> Thanks for your questions!
> 
>     1. "data collection process involves comparing raw noise and golden noise"
> 
> 
> We perform this operation in :
> <img alt="Image" width="1271" height="818" src="https://private-user-images.githubusercontent.com/106738329/477407327-7a9bb45f-9819-4e9c-99f0-75be8d1bc033.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUwNjk4MDAsIm5iZiI6MTc1NTA2OTUwMCwicGF0aCI6Ii8xMDY3MzgzMjkvNDc3NDA3MzI3LTdhOWJiNDVmLTk4MTktNGU5Yy05OWYwLTc1YmU4ZDFiYzAzMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgxM1QwNzE4MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMDQzZDEwNGQ1Y2UzOWRkODgwMDUzNDlhMTZmNTE3NGU1Y2FjMWRlZWMzMzg1M2U0ZmMzYTA0OWMzZTgyYmM5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ObTZayvYsddN6AhnVGnw4al-_QukAf9IaJuiGarjxIc">
> 
>     2. "the HPSV2 scores in your code are normalized to 1"
> 
> 
> We do not normalize the score, we just x100 for the consistency of performance.

That means that if A gets HPSv2 score of 0.2345, in our paper we present it as 23.45

**Comment by hahally** [comment]
![Image](https://github.com/user-attachments/assets/09cf1510-f378-4277-a00d-827e70603e33)
Why is the score 1 before and after optimization? Is this normal?

**Comment by Klayand** [comment]
Hi, hahally. It seems that the original score and the optimized score are the same. Maybe there are some bugs in the data_collection.py.

Very sorry for that. I will quickly check it and give your response.

**Comment by Klayand** [comment]
Hello, 

I have tested this bug several times, not only on my local machine, but also on my server. It seems the results in my server are all right.

`{"index":` 0, "seed": 619, "caption": "A factory for making fluffy kittens", "original_score_list": [0.3134765625], "optimized_score_list": `[0.311279296875]}`

First, I was wondering maybe this issue is caused by the different version of HPSv2, however, I have checked the HPSv2 repo, in https://github.com/tgxs002/HPSv2/blob/866735ecaae999fa714bd9edfa05aa2672669ee3/hpsv2/img_score.py#L42

```
def score(img_path: Union[list, str, Image.Image], prompt: str, cp: str = None, hps_version: str = "v2.0") -> list:
    
    initialize_model()
    model = model_dict['model']
    preprocess_val = model_dict['preprocess_val']

    # check if the checkpoint exists
    if not os.path.exists(root_path):
        os.makedirs(root_path)
    if cp is None:
        cp = huggingface_hub.hf_hub_download("xswu/HPSv2", hps_version_map[hps_version])
    
    checkpoint = torch.load(cp, map_location=device)
    model.load_state_dict(checkpoint['state_dict'])
    tokenizer = get_tokenizer('ViT-H-14')
    model = model.to(device)
    model.eval()
    
    
    if isinstance(img_path, list):
        result = []
        for one_img_path in img_path:
            # Load your image and prompt
            with torch.no_grad():
                # Process the image
                if isinstance(one_img_path, str):
                    image = preprocess_val(Image.open(one_img_path)).unsqueeze(0).to(device=device, non_blocking=True)
                elif isinstance(one_img_path, Image.Image):
                    image = preprocess_val(one_img_path).unsqueeze(0).to(device=device, non_blocking=True)
                else:
                    raise TypeError('The type of parameter img_path is illegal.')
                # Process the prompt
                text = tokenizer([prompt]).to(device=device, non_blocking=True)
                # Calculate the HPS
                with torch.cuda.amp.autocast():
                    outputs = model(image, text)
                    image_features, text_features = outputs["image_features"], outputs["text_features"]
                    logits_per_image = image_features @ text_features.T

                    hps_score = torch.diagonal(logits_per_image).cpu().numpy()
            result.append(hps_score[0])    
        return result
    elif isinstance(img_path, str):
        # Load your image and prompt
        with torch.no_grad():
            # Process the image
            image = preprocess_val(Image.open(img_path)).unsqueeze(0).to(device=device, non_blocking=True)
            # Process the prompt
            text = tokenizer([prompt]).to(device=device, non_blocking=True)
            # Calculate the HPS
            with torch.cuda.amp.autocast():
                outputs = model(image, text)
                image_features, text_features = outputs["image_features"], outputs["text_features"]
                logits_per_image = image_features @ text_features.T

                hps_score = torch.diagonal(logits_per_image).cpu().numpy()
        return [hps_score[0]]
    elif isinstance(img_path, Image.Image):
        # Load your image and prompt
        with torch.no_grad():
            # Process the image
            image = preprocess_val(img_path).unsqueeze(0).to(device=device, non_blocking=True)
            # Process the prompt
            text = tokenizer([prompt]).to(device=device, non_blocking=True)
            # Calculate the HPS
            with torch.cuda.amp.autocast():
                outputs = model(image, text)
                image_features, text_features = outputs["image_features"], outputs["text_features"]
                logits_per_image = image_features @ text_features.T

                hps_score = torch.diagonal(logits_per_image).cpu().numpy()
        return [hps_score[0]]
    else:
        raise TypeError('The type of parameter img_path is illegal.')
```

It seems that in the hpsv2 function, there is no normalization operator. I am confused by this issue.  = . =
And I also check the dataset_collection.py, there is no normalization operator too.

If you have solved this issue, please let me know, thanks!

**Comment by hahally** [comment]
<img width="1885" height="1285" alt="Image" src="https://github.com/user-attachments/assets/66ad3357-a585-46f2-b1e1-96d9883810a2" />
I used HPSV2 from the Google Link you shared. Now I've solved the problem. Thanks!
